migrator/migrations/m_108_to_m_109_replace_deprecated_resources/migration.go

```diff
@@ -0,0 +1,149 @@
+	Indicator             = "Indicator"
+	Licenses              = "Licenses"
+	NetworkBaseline       = "NetworkBaseline"
+	NetworkGraphConfig    = "NetworkGraphConfig"
+	Notifier              = "Notifier"
+	ProbeUpload           = "ProbeUpload"
+	ProcessWhitelist      = "ProcessWhitelist"
+	Risk                  = "Risk"
+	Role                  = "Role"
+	ScannerBundle         = "ScannerBundle"
+	ScannerDefinitions    = "ScannerDefinitions"
+	SensorUpgradeConfig   = "SensorUpgradeConfig"
+	ServiceIdentity       = "ServiceIdentity"
+	SignatureIntegration  = "SignatureIntegration"
+	User                  = "User"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 108,
+		VersionAfter:   storage.Version{SeqNum: 109},
+		Run: func(databases *types.Databases) error {
+			return migatePermissionSets(databases.RocksDB)
+		},
+	}
+
+	prefix = []byte("permission_sets")
+
+	replacements = map[string]string{
+		AllComments:           Administration,
+		APIToken:              Integration,
+		AuthProvider:          Access,
+		BackupPlugins:         Integration,
+		ComplianceRuns:        Compliance,
+		ComplianceRunSchedule: Administration,
+		Config:                Administration,
+		DebugLogs:             Administration,
+		Group:                 Access,
+		ImageComponent:        Image,
+		ImageIntegration:      Integration,
+		Indicator:             DeploymentExtension,
+		Licenses:              Access,
+		NetworkBaseline:       DeploymentExtension,
+		NetworkGraphConfig:    Administration,
+		Notifier:              Integration,
+		ProbeUpload:           Administration,
+		ProcessWhitelist:      DeploymentExtension,
+		Risk:                  DeploymentExtension,
+		Role:                  Access,
+		ScannerBundle:         Administration,
+		ScannerDefinitions:    Administration,
+		SensorUpgradeConfig:   Administration,
+		ServiceIdentity:       Administration,
+		SignatureIntegration:  Integration,
+		User:                  Access,
+	}
+
+	readOpts  = gorocksdb.NewDefaultReadOptions()
+	writeOpts = gorocksdb.NewDefaultWriteOptions()
+)
+
+func propagatePermission(resource string, accessLevel storage.Access, permissions map[string]storage.Access) {
+	if _, found := permissions[resource]; !found {
+		permissions[resource] = accessLevel
+	} else {
+		oldLevel := permissions[resource]
+		if accessLevel > oldLevel {
+			permissions[resource] = accessLevel
+		}
+	}
+}
+
+func migatePermissionSets(db *gorocksdb.DB) error {
+	it := db.NewIterator(readOpts)
+	defer it.Close()
+	wb := gorocksdb.NewWriteBatch()
+	for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
+		permissions := &storage.PermissionSet{}
+		if err := proto.Unmarshal(it.Value().Data(), permissions); err != nil {
+			return errors.Wrap(err, "unable to unmarshal permission set")
+		}
+		// Copy the permission set, removing the deprecated resource permissions, and keeping the
+		// highest access level between that of deprecated resource and their replacement
```

Comment:
I think as a security product we should choose lowest (and call this out in the changelog).